### PR TITLE
Can anybody write test case for this Bayesian LSTM?

### LINKED ISSUE #335: Implementation Request: dropout between gates in LSTM
http://arxiv.org/abs/1512.05287 shows the usage of dropout between gates in LSTM. It would be great if we can also easily use such "inside" dropouts in rnn modules like FastLSTM.
Keras already supports it https://keras.io/layers/recurrent/

### PR DESCRIPTION
**Pull request by jnhwkim**
though this code is used in my recent work.

I implement Bayesian LSTM and can be applied TrimZero, which resolves #335.

### CODE REVIEW

**Comment by nicholas-leonard** [comment]
Don't have to write it. Current test past, so its fine for now.

**Comment by hashbangCoder** [comment]
Hi,

I have been testing the dropout, specifically for `FastLSTM` and there seems to be some implementation errors.

```lua
 p= 0.5    --dropout prob
lstm = nn.LSTM(10,20,nil,nil,p)
print(lstm.p)   --returns 0.5 and works as expected

-- But,
f_lstm = nn.FastLSTM(10,20,nil,nil,nil,nil,p)
print(f_lstm.p)   -- prints 0 !
```
I think the error is in calling the `__init()` of parent class in `FastLSTM` [here](https://github.com/Element-Research/rnn/blob/master/FastLSTM.lua#L34). This overwrites the `self.p` of child class.
If you comment that line for example (or preferably pass the `p` argument to `__init`), it works fine.

Just to be sure, I updated the module before checking. And I checked multiple times for argument positional errors. I  hope I'm not wrong on the OOP part.

@nicholas-leonard  @jnhwkim

**Comment by jnhwkim** [comment]
@hashbangCoder I'll check that shortly. Thanks!

**Comment by tastyminerals** [comment]
@jnhwkim any news on Dropout implementation? Also does there are some issue with GRU dropout as well #415.
